ui/imports/utils/Global.qml

```diff
@@ -5,121 +5,74 @@ import QtQml 2.14
     property var errorSound
 
-    signal openImagePopup(var image, var contextMenu)
     signal openLinkInBrowser(string link)
     signal openChooseBrowserPopup(string link)
-    signal openDownloadModalRequested(bool available, string version, string url)
     signal settingsLoaded()
-    signal openBackUpSeedPopup()
     signal openCreateChatView()
     signal closeCreateChatView()
 
-    signal openProfilePopupRequested(string publicKey, var parentPopup)
-
-    signal openNicknamePopupRequested(string publicKey, string nickname, string subtitle)
-    signal nickNameChanged(string publicKey, string nickname)
-
     signal blockContactRequested(string publicKey, string contactName)
-    signal contactBlocked(string publicKey)
     signal unblockContactRequested(string publicKey, string contactName)
-    signal contactUnblocked(string publicKey)
 
-    signal openChangeProfilePicPopup(var cb)
     signal displayToastMessage(string title, string subTitle, string icon, bool loading, int ephNotifType, string url)
-    signal openEditDisplayNamePopup()
-    signal openActivityCenterPopupRequested
 
+    signal openPopupRequested(var popupComponent, var params);
+    signal openNicknamePopupRequested(string publicKey, string nickname, string subtitle)
+    signal openDownloadModalRequested(bool available, string version, string url)
+    signal openChangeProfilePicPopup(var cb)
+    signal openBackUpSeedPopup()
+    signal openImagePopup(var image, var contextMenu)
+    signal openProfilePopupRequested(string publicKey, var parentPopup)
+    signal openEditDisplayNamePopup()
+    signal openActivityCenterPopupRequested()
+    signal openSendIDRequestPopup(string publicKey, var cb)
     signal openContactRequestPopup(string publicKey, var cb)
-
     signal openInviteFriendsToCommunityPopup(var community, var communitySectionModule, var cb)
-
-    signal openSendIDRequestPopup(string publicKey, var cb)
-
     signal openIncomingIDRequestPopup(string publicKey, var cb)
-
     signal openOutgoingIDRequestPopup(string publicKey, var cb)
 
+    signal openLink(string link)
+
+    signal setNthEnabledSectionActive(int nthSection)
+    signal appSectionBySectionTypeChanged(int sectionType, int subsection)
+
     function openProfilePopup(publicKey, parentPopup) {
-        openProfilePopupRequested(publicKey, parentPopup)
+        root.openProfilePopupRequested(publicKey, parentPopup)
     }
 
     function openActivityCenterPopup() {
-        openActivityCenterPopupRequested()
+        root.openActivityCenterPopupRequested();
     }
 
     function openPopup(popupComponent, params = {}) {
-        const popup = popupComponent.createObject(root.appMain, params)
-        popup.open()
-        return popup
+        root.openPopupRequested(popupComponent, params);
     }
 
     function openDownloadModal(available, version, url){
-        openDownloadModalRequested(available, version, url);
+        root.openDownloadModalRequested(available, version, url);
```

Comment:
this method seems to be not needed at all as it only calls signal

Reply:
@micieslak it is needed, see in AppMain.qml line 101